data_management/access_copy_converter/tests/test_access_copy_converter.py

```diff
@@ -0,0 +1,255 @@
+    assert converter.already_converted("bucket", "key") is True
+    s3.head_object.assert_called_once_with(Bucket="bucket", Key="key")
+
+
+def test_already_converted_not_found(patch_boto_clients):
+    s3 = patch_boto_clients["s3"]
+    err = ClientError({"Error": {"Code": "404"}}, "HeadObject")
+    s3.head_object.side_effect = err
+    assert converter.already_converted("bucket", "key") is False
+
+
+def test_already_converted_other_error_raises(patch_boto_clients):
+    s3 = patch_boto_clients["s3"]
+    err = ClientError({"Error": {"Code": "500"}}, "HeadObject")
+    s3.head_object.side_effect = err
+    with pytest.raises(ClientError):
+        converter.already_converted("bucket", "key")
+
+
+def test_get_extension_from_ffid(sqlite_conn):
+    conn, metadata, ffid, file_table = sqlite_conn
+    conn.execute(insert(ffid).values(FileId="file123", Extension="DOCX"))
+    conn.commit()
+    ext = converter.get_extension("file123", conn, metadata)
+    assert ext == "docx"
+
+
+def test_get_extension_from_filename(sqlite_conn):
+    conn, metadata, ffid, file_table = sqlite_conn
+    conn.execute(
+        insert(file_table).values(FileId="file123", FileName="report.PDF")
+    )
+    conn.commit()
+    ext = converter.get_extension("file123", conn, metadata)
+    assert ext == "pdf"
+
+
+def test_get_extension_no_extension_and_warning(sqlite_conn, caplog):
+    conn, metadata, ffid, file_table = sqlite_conn
+    conn.execute(
+        insert(file_table).values(FileId="file123", FileName="noextfile")
+    )
+    conn.commit()
+    caplog.set_level("WARNING")
+    ext = converter.get_extension("file123", conn, metadata)
+    assert ext is None
+    assert "No extension in FileName" in caplog.text
+
+
+def test_get_extension_ffid_query_failure(monkeypatch, sqlite_conn):
+    conn, metadata, ffid, file_table = sqlite_conn
+    orig_execute = conn.execute
+
+    def execute_raise(stmt, *args, **kwargs):
+        s = str(stmt).lower()
+        if "ffidmetadata" in s:
+            raise SQLAlchemyError("ffid error")
+        return orig_execute(stmt, *args, **kwargs)
+
+    monkeypatch.setattr(conn, "execute", execute_raise)
+    with pytest.raises(Exception) as exc:
+        converter.get_extension("file123", conn, metadata)
+    assert "Error querying FFIDMetadata table" in str(exc.value)
+
+
+def test_get_extension_file_query_failure(monkeypatch, sqlite_conn):
+    conn, metadata, ffid, file_table = sqlite_conn
+
+    def execute_raise(stmt, *args, **kwargs):
+        s = str(stmt).lower()
+
+        if "ffidmetadata" in s:
+            return type("R", (), {"first": lambda self=None: None})()
+
+        raise SQLAlchemyError("file error")
+
+    monkeypatch.setattr(conn, "execute", execute_raise)
+    with pytest.raises(Exception) as exc:
+        converter.get_extension("file123", conn, metadata)
+    assert "Error querying File table" in str(exc.value)
+
+
+def test_convert_with_libreoffice_success(monkeypatch, tmp_path):
```

Comment:
Can we split these tests into sections so its easy to see where were testing different functions

can group tests for a particular function in a class like 

`class TestConvertWithLibreOffice`